java/kotlin-extractor2/src/main/kotlin/entities/Class.kt

```diff
@@ -40,8 +41,13 @@ fun KotlinFileExtractor.extractClassSource(
                     kind != KaClassKind.CLASS &&
                     kind != KaClassKind.OBJECT //&&
                 //OLD KE1: kind != ClassKind.ENUM_ENTRY
-                ) {
-                    logger.warnElement("Unrecognised class kind $kind", c.psiSafe() ?: TODO())
+                ) else {
+                    c.psiSafe<KtElement>().let { e ->
+                        if (e != null)
+                            logger.warnElement("Unrecognised class kind $kind", e)
+                        else
+                            logger.warn("Unrecognised class kind $kind")
+                    }
```

Comment:
A few thoughts on this:

No matter what, I think we should wrap whatever pattern we end up using in a `logger.warnSymbol` method.

Even when there isn't a `psiSafe`, I think we can still get at least a name from the symbol.

I'm not sure if using the location of `psiSafe` is going to be useful or not; we're not actually extracting the source at this point, so is giving the source location going to send us in the wrong direction?